### PR TITLE
[docs] Clarify cluster and image requirements

### PR DESCRIPTION
**Pull request by aravindhp**
Clearly call out when and why the custom VXLAN port cluster configuration should be used and how it affects the image picked in the MachineSet.

### CODE REVIEW

File: docs/vsphere-prerequisites.md [hunk 1]
```diff
@@ -1,14 +1,19 @@
...
 VM. These steps are an overview of the required configuration of the
 golden image.
 
 ## vSphere Windows VM golden image creation
 * Create the VM from an updated version of Windows Server 1909 VM image
-which includes patch
-[KB4565351](https://support.microsoft.com/en-us/help/4565351/windows-10-update-kb4565351)
+  that includes the patch [KB4565351](https://support.microsoft.com/en-us/help/4565351/windows-10-update-kb4565351).
+  This is required to get the OS feature that allows setting the VXLAN UDP port.
+  Please note that this patch is not available for `Windows Server 2019`.
```
Comment (christianh814) [inline docs/vsphere-prerequisites.md:16]: So this is a bit confusing. Are we saying...

* User Windows Server 2019 with this patch
* This patch isn't available for Windows Server 2019

...if so then there's no way for the user to be in a supported state :)

If we mean something different we should clarify

Comment (aravindhp) [reply docs/vsphere-prerequisites.md:16]: I am saying using **1909** :-)

Comment (christianh814) [reply docs/vsphere-prerequisites.md:16]: Nevermind. I see now you're talking about 1909.

Comment (Anandnatraj) [reply docs/vsphere-prerequisites.md:16]: LGTM

File: README.md [hunk 2]
```diff
@@ -58,8 +58,12 @@ The following template variables need to be replaced as follows with values from
 * *\<vCenter Datastore Name\>*: datastore name
 * *\<vCenter Server FQDN/IP\>*: IP address or FQDN of the vCenter server
 
-Please note that on vSphere, Windows Machine names cannot be more than 15 characters long. The MachineSet name therefore
-cannot be more than 9 characters long, due to the way Machine names are generated from it.
+*IMPORTANT*:
+- The template used in the MachineSet has to be a Windows Server 1909
+  image as described in [vSphere prerequisites](docs/vsphere-prerequisites.md).
+- on vSphere, Windows Machine names cannot be more than 15 characters long. The
```
Comment (codyhoag) [inline README.md:64]: s/on vSphere/On vSphere

File: README.md [hunk 3]
```diff
@@ -58,8 +58,12 @@ The following template variables need to be replaced as follows with values from
...
+  MachineSet name therefore cannot be more than 9 characters long, due to the
```
Comment (codyhoag) [inline README.md:65]: s/name therefore cannot/name, therefore, cannot

File: docs/setup-hybrid-OVNKubernetes-cluster.md [hunk 4]
```diff
@@ -49,6 +49,34 @@ spec:
           hostPrefix: 23
 status: {}
 ```
+The above configuration is recommended for AWS and Azure clusters.
+
+For vSphere clusters, you have to add the `hybridOverlayVXLANPort` option to work around the pod-to-pod connectivity
```
Comment (codyhoag) [inline docs/setup-hybrid-OVNKubernetes-cluster.md:54]: s/have to/must

File: docs/vsphere-prerequisites.md [hunk 5]
```diff
@@ -1,14 +1,19 @@
 # vSphere pre-requisites
 
+The vSphere cluster has to be configured with [hybrid OVN Kubernetes networking](docs/setup-hybrid-OVNKubernetes-cluster.md)
```
Comment (codyhoag) [inline docs/vsphere-prerequisites.md:3]: s/has to be/must be

**Review by codyhoag — COMMENTED** [review]
Just some nits from my end. We'll need to figure out the Windows Server images we're officially supporting, as this is recommending 1909 whereas our current docs are only recommending 2019. https://docs.openshift.com/container-platform/4.7/windows_containers/windows-containers-release-notes-2-x.html#wmco-2-0-0

**Review by codyhoag — APPROVED** [review]
/lgtm

File: docs/machineset-aws.md [hunk 6]
```diff
@@ -1,7 +1,10 @@
 # Creating an AWS Windows MachineSet
 
-_\<windows_container_ami\>_ should be replaced with the AMI ID of a Windows image with a container run-time installed. 
- You must use Windows Server 2019 with a version 10.0.17763.1457 or earlier.
+_\<windows_container_ami\>_ should be replaced with the AMI ID of a Windows
+image with a container run-time installed. You must use Windows Server 2019 with
```
Comment (sebsoto) [inline docs/machineset-aws.md:4]: Any container runtime or docker?

Comment (aravindhp) [reply docs/machineset-aws.md:4]: Sure I can add Docker,

**Pull request merged**